argh/tests/help_json_tests.rs

```diff
@@ -0,0 +1,640 @@
+"notes": "",
+"error_codes": [],
+"subcommands": [{"name": "one", "description": "First subcommand."},
+{"name": "two", "description": "Second subcommand."}]
+}
+"###,
+    );
+
+    assert_help_json_string::<TopLevel>(
+        vec!["one", "--help-json"],
+        r###"{
+"usage": "test_arg_0 one --x <x>",
+"description": "First subcommand.",
+"options": [{"short": "", "long": "--x", "description": "how many x"},
+{"short": "", "long": "--help", "description": "display usage information"},
+{"short": "", "long": "--help-json", "description": "display usage information encoded in JSON"}],
+"positional": [],
+"examples": "",
+"notes": "",
+"error_codes": [],
+"subcommands": []
+}
+"###,
+    );
+}
+
+#[test]
+fn help_json_test_multiline_doc_comment() {
+    #[derive(FromArgs)]
+    /// Short description
+    struct Cmd {
+        #[argh(switch)]
+        /// a switch with a description
+        /// that is spread across
+        /// a number of
+        /// lines of comments.
+        _s: bool,
+    }
+    assert_help_json_string::<Cmd>(
+        vec!["--help-json"],
+        r###"{
+"usage": "test_arg_0 [--s]",
+"description": "Short description",
+"options": [{"short": "", "long": "--s", "description": "a switch with a description that is spread across a number of lines of comments."},
+{"short": "", "long": "--help", "description": "display usage information"},
+{"short": "", "long": "--help-json", "description": "display usage information encoded in JSON"}],
+"positional": [],
+"examples": "",
+"notes": "",
+"error_codes": [],
+"subcommands": []
+}
+"###,
+    );
+}
+
+#[test]
+fn help_json_test_basic_args() {
+    #[allow(dead_code)]
+    #[derive(FromArgs)]
+    /// Basic command args demonstrating multiple types and cardinality. "With quotes"
+    struct Basic {
+        /// should the power be on. "Quoted value" should work too.
+        #[argh(switch)]
+        power: bool,
+
+        /// option that is required because of no default and not Option<>.
+        #[argh(option, long = "required")]
+        required_flag: String,
+
+        /// optional speed if not specified it is None.
+        #[argh(option, short = 's')]
+        speed: Option<u8>,
+
+        /// repeatable option.
+        #[argh(option, arg_name = "url")]
+        link: Vec<String>,
+    }
+    assert_help_json_string::<Basic>(
+        vec!["--help-json"],
+        r###"{
+"usage": "test_arg_0 [--power] --required <required> [-s <speed>] [--link <url...>]",
+"description": "Basic command args demonstrating multiple types and cardinality. \"With quotes\"",
+"options": [{"short": "", "long": "--power", "description": "should the power be on. \"Quoted value\" should work too."},
+{"short": "", "long": "--required", "description": "option that is required because of no default and not Option<>."},
+{"short": "s", "long": "--speed", "description": "optional speed if not specified it is None."},
+{"short": "", "long": "--link", "description": "repeatable option."},
```

Comment:
I think the arg_name "url" should be in the JSON output somewhere. I would expect the JSON to not drop any information. Preferably, "url" would be a value of some field.

Along the same lines, the fact that this is a repeatable argument is buried in the human readable description, but I think the schema should accommodate the structure of the argument, in this case, either including a field denoting that this is repeatable (or denoting it can have zero or more values) vs. optional vs. required, something like that. (I'm not sure if argh has an option for "1 or more", i.e., required and repeatable.)

Reply:
Added optionality and arg_name to the JSON.